### DISCUSSION TITLE
Is it possible to release self-contained dll?

### DISCUSSION THREAD

**Discussion by LarrysGIT** [empira/PDFsharp#133] [Q&A] [answered]
Currently I am attempting to use pdfsharp in my powershell script for some quick task, I download pdfsharp from nuget with all other references, and add each dll in my script. it feels a little pain.

![image](https://github.com/empira/PDFsharp/assets/7692156/7a3be37a-d549-4ab1-a3e4-246df56fae3d)

The idea is, is it possible to release pdfsharp with the following options or anything similar? thank you for your work on pdfsharp.

![image](https://github.com/empira/PDFsharp/assets/7692156/e3131f06-be22-4d0b-9677-98c8d99517f5)

**Comment by TH-Soft** [comment]
"Trim unused code" is known to lead to problems with PDFsharp because reflection is used.

Feel free to download the PDFsharp source code and create such a self-contained DLL. Please share your results.

Or maybe share your PowerShell sample so others don't have to start from scratch.

**Comment by ThomasHoevel** [comment, marked as answer]
With the current version (6.2.0 preview 3) there should be no issues with trimming.
